lib/password-strength.js

```diff
@@ -397,6 +399,72 @@
 
 
         /**
+         * Calculates the NIST score based on the following rules from NIST
+         * Special Publication 800-63 as of June 2004.
+         *
+         * * The entropy of the first character is four bits;
+         * * The entropy of the next seven characters are two bits per
+         *   character;
+         * * The ninth through the twentieth character has 1.5 bits of
+         *   entropy per character;
+         * * Characters 21 and above have one bit of entropy per character.
+         * * A "bonus" of six bits is added if both upper case letters and
+         *   non-alphabetic characters are used.
+         * * A "bonus" of six bits is added for passwords of length 1 through
+         *   19 characters following an extensive dictionary check to ensure
+         *   the password is not contained within a large dictionary.
+         *   Passwords of 20 characters or more do not receive this bonus
+         *   because it is assumed they are pass-phrases consisting of
+         *   multiple dictionary words.
+         *
+         * According to Wikipedia, this score might not be overly useful.
+         *
+         * > Later research into human-selected password entropy using newly
+         * > available real world data has demonstrated that the NIST scheme
+         * > does not provide a valid metric for entropy estimation of
+         * > human-selected passwords.
+         *
+         * @param {string} password
+         * @return {number}
+         */
+        this.nistScore = function (password) {
+            var len, score;
+
+            len = password.length;
+            score = 0;
+
+            if (len > 20) {
+                score += len - 20;
+                len = 20;
+            }
+
+            if (len > 8) {
+                score += 1.5 * (len - 8);
+                len = 8;
+            }
+
+            if (len > 1) {
+                score += 2 * (len - 1);
+                len = 1;
+            }
+
+            if (len) {
+                score += 4;
+            }
+
+            /*jslint regexp:true*/
+            if (password.match(/[A-Z]/) && password.match(/[^A-Za-z]/)) {
```

Comment:
Should we have a test for this to make sure things are working properly? Looks like we only have a few tests for NIST.